app/src/main/java/ch/epfl/sweng/favors/FavorDetailView.java

```diff
@@ -0,0 +1,119 @@
+
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Activities that contain this fragment must implement the
+ * to handle interaction events.
+ * Use the {@link FavorDetailView#newInstance} factory method to
+ * create an instance of this fragment.
+ */
+public class FavorDetailView extends android.support.v4.app.Fragment  {
+
+    private static final String TAG = "FAVOR_DETAIL_FRAGMENT";
+
+    public ObservableField<String> title;
+    public ObservableField<String> description;
+    public ObservableField<Integer> tokenCost;
+
+
+    FragmentFavorDetailViewBinding binding;
+
+
+    // TODO: Rename parameter arguments, choose names that match
+    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+    private static final String FAVOR_ID = "favorID";
+
+    private Favor currentFavor;
+
+
+    public FavorDetailView() {
+        // Required empty public constructor
+    }
+
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     *
+     * @param favorID ID of the favor that will be used to display details about.
+     * @return A new instance of fragment FavorDetailView.
+     */
+    // TODO: Rename and change types and number of parameters
+    public static FavorDetailView newInstance(String favorID) {
+        FavorDetailView fragment = new FavorDetailView();
+        Bundle args = new Bundle();
+        args.putString(FAVOR_ID, favorID);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        SharedViewFavor model = ViewModelProviders.of(getActivity()).get(SharedViewFavor.class);
+        model.getfavor().observe(this, newFavor -> {
+            title = newFavor.getObservableStringObject(Favor.StringFields.title);
+            description = newFavor.getObservableStringObject(Favor.StringFields.description);
+            //TODO add token cost binding with new database implementation
+            //tokenCost = new ObservableField<>(newFavor.get(Favor.IntegerFields.))
+        });
+    }
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_favor_detail_view,container,false);
+        binding.setElements(this);
+
+        binding.favReportAbusiveAdd.setOnClickListener((l)->{
+            Toast.makeText(this.getContext(), "issue has been reported! Sorry for the inconvenience", Toast.LENGTH_LONG).show();
+        });
+
+        binding.favIntrestedButton.setOnClickListener((l)->{
+            Toast.makeText(this.getContext(), "We will inform the poster of the add that you are intrested to help!", Toast.LENGTH_LONG).show();
+        });
+
+        return binding.getRoot();
+    }
+
+
+
+    @Override
+    public void onDetach() {
```

Comment:
Here I should put some stuff into a savedInstance. Have not yet looked into this. This is a recurring issue through the entire program. Not only here.